src/base_model.py

```diff
@@ -26,6 +27,8 @@ class Schema(marsh.Schema):
         """The marshmallow schema all model schemas have to inherit from."""
 
         id = marsh.fields.UUID()
+        created_at = marsh.fields.DateTime(format='iso')
```

Comment:
Discussion: Should somebody be able to set updated_at and created_at? 🤔

Reply:
Readonly access to `updated_at` and `created_at` during (de)serialization would be a good idea. Unfortunately marshmallow doesn't support this feature natively. But there are workarounds. Let me try something. I will be back 😎

Reply:
Thanks for the insights 👍 I'm looking forward to your little experiment ;)

Reply:
Update: There is functionality for this but it will just ignore the fields when given in a dict. 

Do we want an exception to be raised in this case? If yes, we need a custom construct (which I already have an idea for).

Reply:
> Do we want an exception to be raised in this case? If yes, we need a custom construct (which I already have an idea for).

I'm personally not a fan of over-configuration of a tool 😇 

> Update: There is functionality for this but it will just ignore the fields when given in a dict.

But I'm a little fan of this 😄 Can you send me a link to the docs please 😇 I'm really interested

Reply:
Here are the docs for readonly and writeonly fields for marshmallow:
https://marshmallow.readthedocs.io/en/stable/quickstart.html#read-only-and-write-only-fields
It is not that much documentation to be honset.

This post deals with a different topic but uses this functionality and states that fields with `dump_only=True` are just ignored during deserialization:
https://stackoverflow.com/questions/70328646/equivalent-of-marshmallow-dump-only-fields-for-pydantic-fastapi-without-multiple

(Also fields with `load_only=True` will be ignored during serialization.)

This post suggests a simple solution for throwing an exception if the dict should contain a readonly field.

What would be the better solution in your oppinion?

Reply:
Thank you very much for the research! 

In my opinion, we should use `dump_only=True` for `created_at` and `updated_at`. This is even used as an example 😄 
<img width="472" alt="Screenshot 2023-04-07 at 12 39 09" src="https://user-images.githubusercontent.com/45163503/230594948-dd894aca-319f-4b06-8035-20be6a2e2288.png">

I'm unsure if we should throw an exception additionally. The case, that somebody tries to define created_at and updated_at by their own, is unlikely. Two safety mechanisms are a bit to much for, aren't they? And I like the simplicity of `dump_only` 😇 

What do you think?

Reply:
I would only have implemented one of the mechanisms. Two safety mechanisms would really be too much. But as `dump_only` is much simpler I tend to use it too.

Reply:
Great 👍 What do you think about adding to to `id` too? Nobody should be able to define or update the id.

Reply:
Sounds good. I will add it.

Update: Well during testing it came out that the stackoverflow post was wrong. When deserialization gets a dump_only-field it raises a `peewee.ValidationError`.

Reply:
> Update: Well during testing it came out that the stackoverflow post was wrong. When deserialization gets a dump_only-field it raises a peewee.ValidationError.

Thanks for the information! 😇

Reply:
Have a look again :)

Reply:
Will accept it after merging dev into this branch :P